torch/optim/asgd.py

```diff
@@ -209,6 +209,11 @@ def _single_tensor_asgd(
     capturable: bool,
     has_complex: bool,
 ):
+    if isinstance(lr, Tensor):
+        if lr.dim() != 0:
+            lr = lr.squeeze()
+            etas = [eta.squeeze() for eta in etas]
```

Comment:
etas should belong in a different if statement

Reply:
https://github.com/pytorch/pytorch/blob/4ece056791d779a6bfb0574c3a26cd6a7e600089/torch/optim/asgd.py#L102-L108

`eta` has the same number of dimensions  as `lr` if `_init_group` is used as usual. If anyone directly uses  `_multi_tensor_asgd` without `_init_group`, we have to code as follows:

```
etas = [eta.squeeze() for eta in etas if isinstance(eta, Tensor) and eta.dim() != 0 else eta]
```

Should I do it?

Reply:
Alternative solution:

```python
state["eta"] = ( 
     torch.as_tensor( 
         _to_scalar(group["lr"]), device=p.device, dtype=_get_scalar_dtype() 
     ) 
     .clone() 
     .detach() 
 ) 
```

Using `_to_scalar` makes `eta` a 0-dim tensor when initializing it.

Reply:
yes this is better

Reply:
I refactored the code: https://github.com/pytorch/pytorch/pull/145674/commits/f796439ab29486b94403e417c3befaf7b75c6e09